---
title: "Figure.3 SDATA-24-03470"
subtitle: "STAR and Kraken stats"
---

```{r include=FALSE}

library(reshape2)
library(ggplot2)
library(ggrepel)
library(cowplot)
#
level.order.treat<-c("siNT","siNRP1.3","siNRP1.5","shNT","shNRP1.3","shNRP1.5") #custom order
#group.colors.treat<-c("darkorange","#E41A1C","purple","black","#0000FF","green")
group.colors.treat<-colorspace::desaturate(c("darkorange","#E41A1C","purple","black","#0000FF","green"),amount=0.15) #muted colors
scales::show_col(group.colors.treat)
#
level.order.line<-c("SUM159","MDAMB231","HS578T") #custom order
group.colors.line<-c("#008000","#0000FF", "#E41A1C")

```


# GC content
```{r}

### GC content after trimming
############################################################
gc<-read.table("U:/Research/Acquisitions/ihbi/apcrcq/rnaseq/OTHER_processed_data/NRP1.v48/results/MultiQC_TrimGalore/multiqc_data/mqc_fastqc_per_sequence_gc_content_plot_Percentages.txt", header=T) #data file generated by the multiqc report
dim(gc) #104 102
colnames(gc) #"total_reads"  - input reads for STAR = total reads after trimming
#
gc<-melt(gc)
head(gc)
gc$variable<-as.numeric(gsub("X","",gc$variable))
#
gc$Line<-gsub("NRP1-","",gc$Sample)
gc$Line<-gsub("-RNA.*","",gc$Line)
gc$Treat<-gsub(".*-","",gc$Line)
gc$Line<-gsub("\\-.*","",gc$Line)
gc$Line<-gsub("\\.","",gc$Line)
head(gc)
#
gc$Treat<-factor(gc$Treat, levels=level.order.treat)
gc$Line<-factor(gc$Line, levels=level.order.line)
#
AA<-ggplot(gc, aes(x=variable, y=value, group=Sample, col=Treat))+
  geom_line(linewidth=0.7)+ 
  #geom_smooth(method = "loess", se = FALSE)+ #NOPE!
  facet_wrap(~Line)+
  theme_bw()+
  scale_color_manual(values=group.colors.treat)+
  theme(legend.title=element_blank())+
  guides(color = guide_legend(override.aes=list(linewidth=2)))+
  labs(title='FastQC: per sequence GC content', x='%GC', y='percentage')
AA
#plotly::ggplotly(GC) #for exploration
  
```



# STAR mapping
```{r}

### raw reads before and after trimming
############################################################
### before trim total reads are the TrimGalore input reads - can use MultiQC / TrimGalore data to extract
raw.reads<-read.table("U:/Research/Acquisitions/ihbi/apcrcq/rnaseq/OTHER_processed_data/NRP1.v48/results/MultiQC_TrimGalore/multiqc_trim.logs.data/multiqc_cutadapt.txt", header=T) 
dim(raw.reads) #104   8
colnames(raw.reads) #"r_processed" - input reads for TrimGalore = total raw reads before any processing
#
### after trim reads are the STAR input reads - can use MultiQC / STAR logs data to extract
star<-read.table("U:/Research/Acquisitions/ihbi/apcrcq/rnaseq/OTHER_processed_data/NRP1.v48/results/STAR/logs/multiqc_data/multiqc_star.txt", header=T)
dim(star) #52 27
colnames(star) #"total_reads"  - input reads for STAR = total reads after trimming
############################################################
#
raw.reads<-raw.reads[grep("_R1$",raw.reads$Sample),c("Sample","r_processed")] #select only the R1 reads, values are the same for R2
dim(raw.reads) #52  2
raw.reads$Sample<-gsub("_R1$","",raw.reads$Sample)
#
identical(raw.reads$Sample, star$Sample)
raw.reads$star.input<-star$total_reads
colnames(raw.reads)<-c("Sample","before.trim","after.trim")
#
raw.reads<-melt(raw.reads)
raw.reads$Line<-gsub("NRP1-","",raw.reads$Sample)
raw.reads$Line<-gsub("-RNA.*","",raw.reads$Line)
raw.reads$Treat<-gsub(".*-","",raw.reads$Line)
raw.reads$Line<-gsub("\\-.*","",raw.reads$Line)
raw.reads$Line<-gsub("\\.","",raw.reads$Line)
head(raw.reads)
#
raw.reads$Treat<-factor(raw.reads$Treat, levels=level.order.treat)
raw.reads$Line<-factor(raw.reads$Line, levels=level.order.line)
raw.reads$value<-raw.reads$value/1000000 #in millions
#
#
# A<-ggplot(raw.reads, aes(x=variable,y=value))+
#    geom_boxplot(lwd=0.4,fatten=2,alpha=0.2,outlier.shape=NA)+
#    geom_point(aes(color=Treat), position=position_jitter(seed=1, width=0.25, height=0.0),size=2,alpha=0.7)+
#    facet_wrap(~Line)+
#    scale_color_manual(values=group.colors.treat)+
#    theme_bw()+
#    theme(legend.title=element_blank())+
#    labs(title='total reads', x=NULL, y='million paired reads')
# A
#
### updated version only showing the STAR input reads
A<-ggplot(raw.reads[raw.reads$variable=='after.trim',], aes(x=Line,y=value))+
   geom_boxplot(lwd=0.4,fatten=2,alpha=0.2,outlier.shape=NA)+
   geom_point(aes(color=Treat), position=position_jitter(seed=1, width=0.25, height=0.0),size=2,alpha=0.7)+
   scale_color_manual(values=group.colors.treat)+
   theme_bw()+
   theme(legend.title=element_blank(), legend.position='none')+
   #guides(color = guide_legend(ncol = 3, byrow = TRUE))+
   theme(axis.text.x=element_text(angle=30, hjust=1))+
   #labs(title='total reads', x=NULL, y='million paired reads')
   labs(x=NULL, y='million paired reads - after trimming')
A
Agrob<-ggplotGrob(A) #make inset

```


```{r}

### star mapping stats - percentages
############################################################
star.pct<-star[,grep("_percent",colnames(star))]
rownames(star.pct)<-star$Sample
dim(star.pct)  #52  6
colnames(star.pct)
rowSums(star.pct)  #sanity check - ok
#
table(star.pct$multimapped_toomany_percent) #all zero! can be removed
star.pct$multimapped_toomany_percent<-NULL
#
star.pct$unmapped_percent<-star.pct$unmapped_mismatches_percent + star.pct$unmapped_tooshort_percent + star.pct$unmapped_other_percent #combine into one unmapped category
star.pct$unmapped_mismatches_percent<-NULL
star.pct$unmapped_tooshort_percent<-NULL
star.pct$unmapped_other_percent<-NULL
rowSums(star.pct)  #sanity check - ok
#
colnames(star.pct)<-gsub("_percent","",colnames(star.pct)) #more concise labels
star.pct<-melt(as.matrix(star.pct), varnames=c("Sample","Type"))
#
star.pct$Line<-gsub("NRP1-","",star.pct$Sample)
star.pct$Line<-gsub("-RNA.*","",star.pct$Line)
star.pct$Treat<-gsub(".*-","",star.pct$Line)
star.pct$Line<-gsub("\\-.*","",star.pct$Line)
star.pct$Line<-gsub("\\.","",star.pct$Line)
head(star.pct)
#
star.pct$Treat<-factor(star.pct$Treat, levels=level.order.treat)
star.pct$Line<-factor(star.pct$Line, levels=level.order.line)
# 
# ggplot(star.pct, aes(x=value, y=sample, fill=type))+
#   geom_bar(stat="identity", position = position_stack(reverse = TRUE))+
#   scale_fill_manual(values=star.colors)+
#   theme_minimal()+
#   theme(legend.title=element_blank())+
#   labs(x='read percentage', y=NULL)
#
#
B<-ggplot(star.pct, aes(x=Line,y=value))+
   geom_boxplot(lwd=0.4,fatten=2,alpha=0.2,outlier.shape=NA)+
   geom_point(aes(color=Treat), position=position_jitter(seed=1, width=0.25, height=0.0),size=2,alpha=0.7)+
   facet_wrap(~Type, scales='free')+
   scale_color_manual(values=group.colors.treat)+
   theme_bw()+
   theme(legend.title=element_blank())+
   theme(axis.text.x=element_text(angle=30, hjust=1))+
   guides(color = guide_legend(override.aes=list(size=3)))+
   labs(title='STAR mapping statistics - summarised', x=NULL, y='percent of trimmed reads')
B

```

```{r}

### star mapping stats - number of reads
############################################################
star.reads<-star[,c("uniquely_mapped","multimapped","multimapped_toomany","unmapped_mismatches","unmapped_tooshort","unmapped_other")]
rownames(star.reads)<-star$Sample
star.colors<-c('#3176b0','#7bb2e0','#f7a053','#e5178c','#ad1b47','#6c0d0e') #picked from star multiqc report
#
star.reads<-melt(as.matrix(star.reads), varnames=c("sample","type"))
star.reads$value<-star.reads$value/1000000 #in millions
head(star.reads)
star.reads$sample.short<-sub("^NRP1-","",star.reads$sample) #shorten the sample names
#
C<-ggplot(star.reads, aes(x=value, y=sample.short, fill=type))+
  geom_bar(stat="identity", position = position_stack(reverse = TRUE), width=0.7)+
  scale_fill_manual(values=star.colors, labels = function(x) paste0(x, "     "))+ #add a bit more space between labels
  theme_minimal()+
  theme(plot.margin = unit(c(10, 10, 10, 10), "pt"))+ #default is 5.5
  theme(legend.title=element_blank(), legend.position='bottom')+
  #theme(legend.key.spacing.x = unit(1, "cm"))+ #needs ggplot2>3.5.0
  theme(legend.key.size = unit(0.9, "lines"))+
  guides(fill = guide_legend(ncol = 3, byrow = TRUE))+ #override.aes doesn't work for bars
  geom_hline(yintercept=c(17.5,35.5))+ 
  scale_x_continuous(limits=c(0,100))+ #extend the x axis a bit
  annotation_custom(Agrob, xmin=80, xmax=105, ymin=22, ymax=53)+ #plot A as inset
  labs(title='STAR mapping statistics - per sample', x='million paired reads - after trimming', y=NULL)
C

```



# Kraken
```{r}

### data file of 'topfive' exported from multiqc report 
kraken.domain<-read.csv("U:/Research/Projects/ihbi/apcrcq/rnaseq/NRP1.v48/pipeline.hpc/kraken-topfive-plot-domain.csv")
dim(kraken.domain)
head(kraken.domain)
#
rownames(kraken.domain)<-gsub("-kraken2-report","",kraken.domain$Category)
identical(rownames(kraken.domain),star$Sample) #TRUE
kraken.domain$total_reads<-star$total_reads
#
kraken.domain.cpm<-kraken.domain[,c('Bacteria','Eukaryota','Viruses','Archaea')]
kraken.domain.cpm<-kraken.domain.cpm/kraken.domain$total_reads*1000000
kraken.domain.cpm<-melt(as.matrix(kraken.domain.cpm))
head(kraken.domain.cpm)   
kraken.domain.cpm$sample.short<-sub("^NRP1-","",kraken.domain.cpm$Var1) #shorten sample names
#
#
kraken.colors<-c('#7cb5ec','#434348','#90ed7d','#f7a35c','#cccccc','#d4949c') #picked from kraken multiqc report
#
D<-ggplot(kraken.domain.cpm, aes(x=value, y=sample.short, fill=Var2))+
  geom_bar(stat="identity", position = position_stack(reverse = TRUE), width=0.7)+
  scale_fill_manual(values=kraken.colors)+
  theme_minimal()+
  theme(plot.margin = unit(c(10, 10, 10, 10), "pt"))+ #default is 5.5
  theme(legend.title=element_blank(), legend.position='bottom')+
  theme(legend.key.size = unit(0.9, "lines"))+
  guides(fill = guide_legend(ncol = 4, byrow = TRUE))+
  geom_hline(yintercept=c(17.5,35.5))+  
  labs(title="Kraken2 - domains", x='cpm', y=NULL)
D

```


```{r include=FALSE}

### microbiome info from the kraken reports
#####################################################
kraken.reports<-list.files(path=paste0("U:/Research/Acquisitions/ihbi/apcrcq/rnaseq/OTHER_processed_data/NRP1.v48/results/Kraken2"), pattern=".*report.txt", full.names=T)
kraken.reports
#
kraken.names<-gsub(".*/","",kraken.reports)
kraken.names<-gsub("-kraken.*","",kraken.names)
kraken.names
#
#
### Mycoplasma
#####################################################
myco.data<-data.frame() #make empty data frame
for (file in kraken.reports){
input<-read.delim(file=file,sep="\t",header=F)  #read the kraken report
input<-input[input$V5==2093,] #spathogen.id ("2093","Mycoplasma")
if(nrow(input)==0){input[1,]<-c(0,0,0,0,pathogen.id,0)} #if there's no counts at all in a sample
myco.data<-rbind(myco.data,input)
}
head(myco.data)
myco.data$Sample<-kraken.names
#
identical(myco.data$Sample,star$Sample) #TRUE
myco.data$total_reads<-star$total_reads
myco.data$cpm<-myco.data$V2/myco.data$total_reads*1000000
myco.data$sample.short<-sub("^NRP1-","",myco.data$Sample) #shorten sample names
#
###myco.data$Line<-rep(c('s','h','t'),times=c(x,x,x))
#
#
E<-ggplot(data=myco.data, aes(x=cpm, y=sample.short))+ #cpm for microbe
  geom_col(width=0.7)+   
  geom_text(aes(label=format(cpm,big.mark=",",digits=1,scientific=F)), hjust=-0.2, vjust=0.4, size=3)+
  theme_minimal()+
  theme(plot.margin = unit(c(10, 10, 10, 10), "pt"))+ #default is 5.5
  scale_x_continuous(limits=c(0,3.1))+ #expand = c(.02, 0, .3, 0))+ #expand axis to make room for labels
  geom_hline(yintercept=c(17.5,35.5))+  
  labs(title="Kraken2 - mycoplasma", x="cpm", y=NULL)
E

```



# Combined
```{r}

### A4 page in inches: 8.3 x 11.7 inches --> width=15,height=21.14
dev.new(width=15, height=17, noRStudioGD = T, rescale='fixed')
#
plot<-plot_grid(plot_grid(plot_grid(AA,B, ncol=1, labels=c('a','b')), plot_grid(C, labels='c'), rel_widths=c(1,1.6)),   
                plot_grid(D,E, rel_widths=c(1.2,1), labels=c('d','e')),
                ncol=1, rel_heights=c(1,1), #panel_spacing = unit(1, "lines"), 
                align="h",axis="tb")
plot
#  
cowplot::save_plot("Figure3.pdf", plot = plot, base_width = 15, base_height = 17)
#
#
# #### alternative version with panels abc rearranged
# plot<-plot_grid(plot_grid(plot_grid(C, labels='a'), plot_grid(AA,B, ncol=1, labels=c('b','c')), rel_widths=c(1.6,1)),   
#                 plot_grid(D,E, rel_widths=c(1.2,1), labels=c('d','e')),
#                 ncol=1, rel_heights=c(1,1), #panel_spacing = unit(1, "lines"), 
#                 align="h",axis="tb")
# plot
# #  
# cowplot::save_plot("Figure3.v2.pdf", plot = plot, base_width = 15, base_height = 17)

```


<br>

***  

# R.session Info


```{r Rsession, echo=FALSE}

sessionInfo() #record the R and package versions used
##############################
# R version 4.2.3 (2023-03-15 ucrt)
# Platform: x86_64-w64-mingw32/x64 (64-bit)
# Running under: Windows 10 x64 (build 19044)
# 
# Matrix products: default
# 
# locale:
# [1] LC_COLLATE=English_Australia.utf8  LC_CTYPE=English_Australia.utf8   
# [3] LC_MONETARY=English_Australia.utf8 LC_NUMERIC=C                      
# [5] LC_TIME=English_Australia.utf8    
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.1.1  ggrepel_0.9.3  ggplot2_3.4.2  reshape2_1.4.4 edgeR_3.40.2   limma_3.54.2  
# 
# loaded via a namespace (and not attached):
#  [1] Rcpp_1.0.10                 locfit_1.5-9.7              lattice_0.20-45            
#  [4] Rsamtools_2.14.0            Biostrings_2.66.0           digest_0.6.31              
#  [7] R6_2.5.1                    GenomeInfoDb_1.34.9         plyr_1.8.8                 
# [10] stats4_4.2.3                Polychrome_1.5.1            evaluate_0.21              
# [13] pillar_1.11.0               zlibbioc_1.44.0             rlang_1.1.3                
# [16] rstudioapi_0.14             S4Vectors_0.36.2            Matrix_1.6-5               
# [19] rmarkdown_2.22              BiocParallel_1.32.5         stringr_1.5.0              
# [22] RCurl_1.98-1.12             DelayedArray_0.23.2         compiler_4.2.3             
# [25] rtracklayer_1.58.0          xfun_0.39                   pkgconfig_2.0.3            
# [28] BiocGenerics_0.44.0         htmltools_0.5.4             tidyselect_1.2.0           
# [31] SummarizedExperiment_1.28.0 tibble_3.2.1                GenomeInfoDbData_1.2.9     
# [34] IRanges_2.32.0              codetools_0.2-19            matrixStats_1.0.0          
# [37] XML_3.99-0.13               crayon_1.5.2                dplyr_1.1.2                
# [40] withr_2.5.0                 GenomicAlignments_1.34.1    bitops_1.0-7               
# [43] grid_4.2.3                  gtable_0.3.3                lifecycle_1.0.3            
# [46] magrittr_2.0.3              scales_1.4.0                cli_3.6.0                  
# [49] stringi_1.7.12              farver_2.1.1                XVector_0.38.0             
# [52] scatterplot3d_0.3-43        generics_0.1.3              vctrs_0.6.0                
# [55] rjson_0.2.21                restfulr_0.0.15             RColorBrewer_1.1-3         
# [58] tools_4.2.3                 Biobase_2.58.0              glue_1.6.2                 
# [61] MatrixGenerics_1.10.0       parallel_4.2.3              fastmap_1.1.1              
# [64] yaml_2.3.7                  colorspace_2.1-0            GenomicRanges_1.50.2       
# [67] knitr_1.43                  BiocIO_1.8.0               
##############################

```

***
