---
title: "Figure.3 SDATA-24-03470"
subtitle: "GSE266566.STAR.and.Kraken2.stats"
---

```{r include=FALSE}

library(reshape2)
library(ggplot2)
library(ggrepel)
library(cowplot)
#
level.order.treat<-c("siNT","siNRP1.3","siNRP1.5","shNT","shNRP1.3","shNRP1.5") #custom order
group.colors.treat<-colorspace::desaturate(c("darkorange","#E41A1C","purple","black","#0000FF","green"),amount=0.15) #muted colors
scales::show_col(group.colors.treat)
#
level.order.line<-c("SUM159","MDAMB231","HS578T") #custom order
group.colors.line<-c("#008000","#0000FF", "#E41A1C")

```


# GC content
```{r}

### GC content after trimming
############################################################
gc<-read.table("./files/mqc_fastqc_per_sequence_gc_content_plot_Percentages.txt", header=T) #data file generated by the multiqc report
dim(gc) #104 102
colnames(gc) #"total_reads"  - input reads for STAR = total reads after trimming
#
gc<-melt(gc)
head(gc)
gc$variable<-as.numeric(gsub("X","",gc$variable))
#
gc$Line<-gsub("NRP1-","",gc$Sample)
gc$Line<-gsub("-RNA.*","",gc$Line)
gc$Treat<-gsub(".*-","",gc$Line)
gc$Line<-gsub("\\-.*","",gc$Line)
gc$Line<-gsub("\\.","",gc$Line)
head(gc)
#
gc$Treat<-factor(gc$Treat, levels=level.order.treat)
gc$Line<-factor(gc$Line, levels=level.order.line)
#
AA<-ggplot(gc, aes(x=variable, y=value, group=Sample, col=Treat))+
  geom_line(linewidth=0.7)+ 
  #geom_smooth(method = "loess", se = FALSE)+ #NOPE!
  facet_wrap(~Line)+
  theme_bw()+
  scale_color_manual(values=group.colors.treat)+
  theme(legend.title=element_blank())+
  guides(color = guide_legend(override.aes=list(linewidth=2)))+
  labs(title='FastQC: per sequence GC content', x='%GC', y='percentage')
AA
#plotly::ggplotly(GC) #for exploration
  
```



# STAR mapping
```{r}

### raw reads before and after trimming
############################################################
### before trim total reads are the TrimGalore input reads - can use MultiQC / TrimGalore data to extract
raw.reads<-read.table("./files/multiqc_cutadapt.txt", header=T) 
dim(raw.reads) #104   8
colnames(raw.reads) #"r_processed" - input reads for TrimGalore = total raw reads before any processing
#
### after trim reads are the STAR input reads - can use MultiQC / STAR logs data to extract
star<-read.table("./files/multiqc_star.txt", header=T)
dim(star) #52 27
colnames(star) #"total_reads"  - input reads for STAR = total reads after trimming
############################################################
#
raw.reads<-raw.reads[grep("_R1$",raw.reads$Sample),c("Sample","r_processed")] #select only the R1 reads, values are the same for R2
dim(raw.reads) #52  2
raw.reads$Sample<-gsub("_R1$","",raw.reads$Sample)
#
identical(raw.reads$Sample, star$Sample)
raw.reads$star.input<-star$total_reads
colnames(raw.reads)<-c("Sample","before.trim","after.trim")
#
raw.reads<-melt(raw.reads)
raw.reads$Line<-gsub("NRP1-","",raw.reads$Sample)
raw.reads$Line<-gsub("-RNA.*","",raw.reads$Line)
raw.reads$Treat<-gsub(".*-","",raw.reads$Line)
raw.reads$Line<-gsub("\\-.*","",raw.reads$Line)
raw.reads$Line<-gsub("\\.","",raw.reads$Line)
head(raw.reads)
#
raw.reads$Treat<-factor(raw.reads$Treat, levels=level.order.treat)
raw.reads$Line<-factor(raw.reads$Line, levels=level.order.line)
raw.reads$value<-raw.reads$value/1000000 #in millions
#
#
### updated version only showing the STAR input reads
A<-ggplot(raw.reads[raw.reads$variable=='after.trim',], aes(x=Line,y=value))+
   geom_boxplot(lwd=0.4,fatten=2,alpha=0.2,outlier.shape=NA)+
   geom_point(aes(color=Treat), position=position_jitter(seed=1, width=0.25, height=0.0),size=2,alpha=0.7)+
   scale_color_manual(values=group.colors.treat)+
   theme_bw()+
   theme(legend.title=element_blank(), legend.position='none')+
   theme(axis.text.x=element_text(angle=30, hjust=1))+
   labs(x=NULL, y='million paired reads - after trimming')
A
Agrob<-ggplotGrob(A) #make inset

```


```{r}

### star mapping stats - percentages
############################################################
star.pct<-star[,grep("_percent",colnames(star))]
rownames(star.pct)<-star$Sample
dim(star.pct)  #52  6
colnames(star.pct)
rowSums(star.pct)  #sanity check - ok
#
table(star.pct$multimapped_toomany_percent) #all zero! can be removed
star.pct$multimapped_toomany_percent<-NULL
#
star.pct$unmapped_percent<-star.pct$unmapped_mismatches_percent + star.pct$unmapped_tooshort_percent + star.pct$unmapped_other_percent #combine into one unmapped category
star.pct$unmapped_mismatches_percent<-NULL
star.pct$unmapped_tooshort_percent<-NULL
star.pct$unmapped_other_percent<-NULL
rowSums(star.pct)  #sanity check - ok
#
colnames(star.pct)<-gsub("_percent","",colnames(star.pct)) #more concise labels
star.pct<-melt(as.matrix(star.pct), varnames=c("Sample","Type"))
#
star.pct$Line<-gsub("NRP1-","",star.pct$Sample)
star.pct$Line<-gsub("-RNA.*","",star.pct$Line)
star.pct$Treat<-gsub(".*-","",star.pct$Line)
star.pct$Line<-gsub("\\-.*","",star.pct$Line)
star.pct$Line<-gsub("\\.","",star.pct$Line)
head(star.pct)
#
star.pct$Treat<-factor(star.pct$Treat, levels=level.order.treat)
star.pct$Line<-factor(star.pct$Line, levels=level.order.line)
# 
#
B<-ggplot(star.pct, aes(x=Line,y=value))+
   geom_boxplot(lwd=0.4,fatten=2,alpha=0.2,outlier.shape=NA)+
   geom_point(aes(color=Treat), position=position_jitter(seed=1, width=0.25, height=0.0),size=2,alpha=0.7)+
   facet_wrap(~Type, scales='free')+
   scale_color_manual(values=group.colors.treat)+
   theme_bw()+
   theme(legend.title=element_blank())+
   theme(axis.text.x=element_text(angle=30, hjust=1))+
   guides(color = guide_legend(override.aes=list(size=3)))+
   labs(title='STAR mapping statistics - summarised', x=NULL, y='percent of trimmed reads')
B

```

```{r}

### star mapping stats - number of reads
############################################################
star.reads<-star[,c("uniquely_mapped","multimapped","multimapped_toomany","unmapped_mismatches","unmapped_tooshort","unmapped_other")]
rownames(star.reads)<-star$Sample
star.colors<-c('#3176b0','#7bb2e0','#f7a053','#e5178c','#ad1b47','#6c0d0e') #picked from star multiqc report
#
star.reads<-melt(as.matrix(star.reads), varnames=c("sample","type"))
star.reads$value<-star.reads$value/1000000 #in millions
head(star.reads)
star.reads$sample.short<-sub("^NRP1-","",star.reads$sample) #shorten the sample names
#
C<-ggplot(star.reads, aes(x=value, y=sample.short, fill=type))+
  geom_bar(stat="identity", position = position_stack(reverse = TRUE), width=0.7)+
  scale_fill_manual(values=star.colors, labels = function(x) paste0(x, "     "))+ #add a bit more space between labels
  theme_minimal()+
  theme(plot.margin = unit(c(10, 10, 10, 10), "pt"))+ #default is 5.5
  theme(legend.title=element_blank(), legend.position='bottom')+
  theme(legend.key.size = unit(0.9, "lines"))+
  guides(fill = guide_legend(ncol = 3, byrow = TRUE))+ #override.aes doesn't work for bars
  geom_hline(yintercept=c(17.5,35.5))+ 
  scale_x_continuous(limits=c(0,100))+ #extend the x axis a bit
  annotation_custom(Agrob, xmin=80, xmax=105, ymin=22, ymax=53)+ #plot A as inset
  labs(title='STAR mapping statistics - per sample', x='million paired reads - after trimming', y=NULL)
C

```



# Kraken
```{r}

### data file of 'topfive' exported from multiqc report 
kraken.domain<-read.csv("./files/kraken-topfive-plot-domain.csv")
dim(kraken.domain)
head(kraken.domain)
#
rownames(kraken.domain)<-gsub("-kraken2-report","",kraken.domain$Category)
identical(rownames(kraken.domain),star$Sample) #TRUE
kraken.domain$total_reads<-star$total_reads
#
kraken.domain.cpm<-kraken.domain[,c('Bacteria','Eukaryota','Viruses','Archaea')]
kraken.domain.cpm<-kraken.domain.cpm/kraken.domain$total_reads*1000000
kraken.domain.cpm<-melt(as.matrix(kraken.domain.cpm))
head(kraken.domain.cpm)   
kraken.domain.cpm$sample.short<-sub("^NRP1-","",kraken.domain.cpm$Var1) #shorten sample names
#
#
kraken.colors<-c('#7cb5ec','#434348','#90ed7d','#f7a35c','#cccccc','#d4949c') #picked from kraken multiqc report
#
D<-ggplot(kraken.domain.cpm, aes(x=value, y=sample.short, fill=Var2))+
  geom_bar(stat="identity", position = position_stack(reverse = TRUE), width=0.7)+
  scale_fill_manual(values=kraken.colors)+
  theme_minimal()+
  theme(plot.margin = unit(c(10, 10, 10, 10), "pt"))+ #default is 5.5
  theme(legend.title=element_blank(), legend.position='bottom')+
  theme(legend.key.size = unit(0.9, "lines"))+
  guides(fill = guide_legend(ncol = 4, byrow = TRUE))+
  geom_hline(yintercept=c(17.5,35.5))+  
  labs(title="Kraken2 - domains", x='cpm', y=NULL)
D

```


```{r include=FALSE}

### microbiome info from the kraken reports
#####################################################
kraken.reports<-list.files(path=paste0("./files"), pattern=".*kraken2-report.txt", full.names=T)
kraken.reports
#
kraken.names<-gsub(".*/","",kraken.reports)
kraken.names<-gsub("-kraken.*","",kraken.names)
kraken.names
#
#
### Mycoplasma
#####################################################
myco.data<-data.frame() #make empty data frame
for (file in kraken.reports){
input<-read.delim(file=file,sep="\t",header=F)  #read the kraken report
input<-input[input$V5==2093,] #spathogen.id ("2093","Mycoplasma")
if(nrow(input)==0){input[1,]<-c(0,0,0,0,pathogen.id,0)} #if there's no counts at all in a sample
myco.data<-rbind(myco.data,input)
}
head(myco.data)
myco.data$Sample<-kraken.names
#
identical(myco.data$Sample,star$Sample) #TRUE
myco.data$total_reads<-star$total_reads
myco.data$cpm<-myco.data$V2/myco.data$total_reads*1000000
myco.data$sample.short<-sub("^NRP1-","",myco.data$Sample) #shorten sample names
#
#
E<-ggplot(data=myco.data, aes(x=cpm, y=sample.short))+ #cpm for microbe
  geom_col(width=0.7)+   
  geom_text(aes(label=format(cpm,big.mark=",",digits=1,scientific=F)), hjust=-0.2, vjust=0.4, size=3)+
  theme_minimal()+
  theme(plot.margin = unit(c(10, 10, 10, 10), "pt"))+ #default is 5.5
  scale_x_continuous(limits=c(0,3.1))+ #expand = c(.02, 0, .3, 0))+ #expand axis to make room for labels
  geom_hline(yintercept=c(17.5,35.5))+  
  labs(title="Kraken2 - mycoplasma", x="cpm", y=NULL)
E

```



# Combined
```{r}

### A4 page in inches: 8.3 x 11.7 inches --> width=15,height=21.14
dev.new(width=15, height=17, noRStudioGD = T, rescale='fixed')
#
plot<-plot_grid(plot_grid(plot_grid(AA,B, ncol=1, labels=c('a','b')), plot_grid(C, labels='c'), rel_widths=c(1,1.6)),   
                plot_grid(D,E, rel_widths=c(1.2,1), labels=c('d','e')),
                ncol=1, rel_heights=c(1,1), #panel_spacing = unit(1, "lines"), 
                align="h",axis="tb")
plot
#  
cowplot::save_plot("Figure3.pdf", plot = plot, base_width = 15, base_height = 17)

```


<br>

***  

# R.session Info


```{r Rsession, echo=FALSE}

sessionInfo() #record the R and package versions used
#############################
# RStudio 2024.12.1+563 "Kousa Dogwood" Release (27771613951643d8987af2b2fb0c752081a3a853, 2025-02-02) for windows
# Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2024.12.1+563 Chrome/126.0.6478.234 Electron/31.7.7 Safari/537.36, Quarto 1.5.57
##############################
# R version 4.4.3 (2025-02-28 ucrt)
# Platform: x86_64-w64-mingw32/x64
# Running under: Windows 11 x64 (build 22631)
# 
# Matrix products: default
# 
# 
# locale:
# [1] LC_COLLATE=English_Australia.utf8  LC_CTYPE=English_Australia.utf8   
# [3] LC_MONETARY=English_Australia.utf8 LC_NUMERIC=C                      
# [5] LC_TIME=English_Australia.utf8    
# 
# time zone: Australia/Brisbane
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.2.0  ggrepel_0.9.6  reshape2_1.4.4 dplyr_1.1.4    ggplot2_4.0.0 
# 
# loaded via a namespace (and not attached):
#  [1] vctrs_0.6.5        cli_3.6.5          knitr_1.50         rlang_1.1.6        xfun_0.53         
#  [6] stringi_1.8.7      generics_0.1.4     S7_0.2.0           glue_1.8.0         colorspace_2.1-1  
# [11] plyr_1.8.9         htmltools_0.5.8.1  rsconnect_1.5.1    scales_1.4.0       rmarkdown_2.29    
# [16] grid_4.4.3         evaluate_1.0.5     tibble_3.3.0       fastmap_1.2.0      yaml_2.3.10       
# [21] lifecycle_1.0.4    stringr_1.5.2      compiler_4.4.3     RColorBrewer_1.1-3 Rcpp_1.1.0        
# [26] pkgconfig_2.0.3    rstudioapi_0.17.1  farver_2.1.2       digest_0.6.37      R6_2.6.1          
# [31] tidyselect_1.2.1   pillar_1.11.0      magrittr_2.0.4     withr_3.0.2        tools_4.4.3       
# [36] gtable_0.3.6  
##############################

```

***